### PR TITLE
Blocking refactor

### PR DESCRIPTION
**Pull request by pxp9**

### CODE REVIEW

**Comment by ayrat555** [comment]
let's use https://github.com/ayrat555/fang/pull/71 as a base branch. in this case the code won't be duplicated

File: src/blocking/runnable.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+use crate::queue::Queueable;
+use crate::Error;
+use chrono::DateTime;
+use chrono::Utc;
+
+pub const COMMON_TYPE: &str = "common";
+
+pub enum Scheduled {
```
Comment (ayrat555) [inline src/blocking/runnable.rs:8]: I think it can be lib.rs because it's shared between blocking and asynk

File: src/blocking/executor.rs [hunk 2]
```diff
@@ -113,32 +83,32 @@ impl Executor {
...
                 Err((task, error)) => {
-                    Queue::fail_task_query(&self.pooled_connection, &task, error).unwrap();
+                    self.queue.fail_task(&task, error).unwrap();
                 }
             },
         }
     }
 }
 
+//TODO fix tests
```
Comment (ayrat555) [inline src/blocking/executor.rs:111]: let's not add such comments. you can create a list of todos in the PR's description

**Review by ayrat555 — APPROVED** [review]

File: src/blocking/executor.rs (hunk 2 above)
Comment (pxp9) [reply src/blocking/executor.rs]: deleted in [Finishing blocking module, starting to modify tests](https://github.com/ayrat555/fang/pull/74/commits/7d1402f52f94e1d0520746e15693116b5988ed4f)

File: src/blocking/runnable.rs (hunk 1 above)
Comment (pxp9) [reply src/blocking/runnable.rs]: [Finishing blocking module, starting to modify tests](https://github.com/ayrat555/fang/pull/74/commits/7d1402f52f94e1d0520746e15693116b5988ed4f)

File: Cargo.toml [hunk 3]
```diff
@@ -12,24 +12,24 @@ rust-version = "1.62"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [features]
-default = ["blocking", "asynk"]
+default = ["blocking"]
```
Comment (ayrat555) [inline Cargo.toml:15]: why did you remove "asynk"?

File: src/blocking/queue.rs [hunk 4]
```diff
@@ -374,499 +352,331 @@ impl Queue {
...
-            let task = insert_new_task(&queue.connection);
-
-            let updated_task = queue.finish_task(&task).unwrap();
-
-            assert_eq!(FangTaskState::Finished, updated_task.state);
-
-            Ok(())
-        });
+    #[derive(Serialize, Deserialize)]
+    struct AyratTask {
```
Comment (ayrat555) [inline src/blocking/queue.rs:396]: 😊

**Review by ayrat555 — APPROVED** [review]

File: src/blocking/queue.rs (hunk 4 above)
Comment (pxp9) [reply src/blocking/queue.rs:396]: :D

File: Cargo.toml (hunk 3 above)
Comment (pxp9) [reply Cargo.toml:15]: For some reason you deleted in your commit i will put again 


image

Comment (pxp9) [reply Cargo.toml]: [asynk feature](https://github.com/ayrat555/fang/pull/74/commits/44babbcea2b98cf2aa241b2d3d249bedf01f6e21)
